build.gradle.kts

```diff
@@ -5,323 +5,325 @@ import kotlin.collections.ArrayList
-            "codesign", "--remove-signature", appDirname
-         ))
-         val processBuilder4 = ProcessBuilder()
-         processBuilder4.command(parameters4)
-         val process4 = processBuilder4.start()
-         if (process4.waitFor() != 0) {
-            throw GradleException("Error while executing codesign --remove-signature")
-         }
+  group = "build"
+  description = "Makes the Mac application"
+  dependsOn("shadowJar", "createDistDir")
+  inputs.dir("$buildDir/libs")
+  inputs.dir("$projectDir/support/jpackage/macos")
+  outputs.dir(ext.get("appDirname") as String)
+  doLast {
+    if (OperatingSystem.current().isMacOsX) {
+      val appDirname = ext.get("appDirname") as String
+      delete(appDirname)
+      val parameters = ArrayList<String>(ext.get("sharedParameters") as ArrayList<String>)
+      parameters.addAll(Arrays.asList(
+          "--name", ext.get("uppercaseProjectName") as String,
+          "--file-associations", "$projectDir/support/jpackage/macos/file.jpackage",
+          "--icon", "$projectDir/support/jpackage/macos/Logisim-evolution.icns",
+          "--type", "app-image"
+      ))
+      val processBuilder1 = ProcessBuilder()
+      processBuilder1.command(parameters)
+      val process1 = processBuilder1.start()
+      if (process1.waitFor() != 0) {
+        throw GradleException("Error while creating app directory")
+      }
+      val pListFilename = "$appDirname/Contents/Info.plist"
+      val parameters2 = ArrayList<String>(Arrays.asList(
+          "awk",
+          "/Unknown/{sub(/Unknown/,\"public.app-category.education\")};"
+              + "{print >\"$buildDir/dist/Info.plist\"};"
+              + "/NSHighResolutionCapable/{"
+                  + "print \"  <string>true</string>\" >\"$buildDir/dist/Info.plist\";"
+                  + "print \"  <key>NSSupportsAutomaticGraphicsSwitching</key>\" >\"$buildDir/dist/Info.plist\""
+              + "}",
+          pListFilename
+      ))
+      val processBuilder2 = ProcessBuilder()
+      processBuilder2.command(parameters2)
+      val process2 = processBuilder2.start()
+      if (process2.waitFor() != 0) {
+        throw GradleException("Error while patching Info.plist")
+      }
+      val parameters3 = ArrayList<String>(Arrays.asList(
+          "mv", "$buildDir/dist/Info.plist", pListFilename
+      ))
+      val processBuilder3 = ProcessBuilder()
+      processBuilder3.command(parameters3)
+      val process3 = processBuilder3.start()
+      if (process3.waitFor() != 0) {
+        throw GradleException("Error while moving Info.plist into app")
       }
-   }
+      val parameters4 = ArrayList<String>(Arrays.asList(
+          "codesign", "--remove-signature", appDirname
+      ))
+      val processBuilder4 = ProcessBuilder()
+      processBuilder4.command(parameters4)
+      val process4 = processBuilder4.start()
+      if (process4.waitFor() != 0) {
+        throw GradleException("Error while executing codesign --remove-signature")
```

Comment:
I'd put `:` after `executing` to separate message from command

Reply:
That is a little more clear.

Reply:
Done.